e2e/test/helpers/templates/FaultInjection.cs

```diff
@@ -227,6 +227,18 @@ public static bool FaultShouldDisconnect(string faultType)
 
                 deviceClient.Dispose();
                 await testDevice.RemoveDeviceAsync().ConfigureAwait(false);
+
+                if (FaultShouldDisconnect(faultType))
```

Comment:
From what I recall, the faults which were hub-wide were the faults that didn't trigger a disconnect, i.e. auth error, quota exceeded and throttling. 
In any case, I think a comment here calling the condition out would help prevent future confusion.

Reply:
I no, I had it backwards. Fixed.